File: python/week3/lesson7/ArrayListCRUDExample.py
students = []

# Create
students.append("Alice")
students.append("Bob")
students.append("Charlie")
students.insert(1, "David")
print("After adding:", students)

# Read
student = students[2]
print("Student at index 2:", student)
exists = "Bob" in students
print("Contains Bob:", exists)
index = students.index("Charlie")
print("Index of Charlie:", index)
print("All students:")

for name in students:
    print(name)

# Update
students[1] = "Emily"
print("After updating:", students)

# Delete by value
students.remove("Bob")
print("After removing Bob:", students)
# Delete by index
students.pop(0)
print("After removing index 0:", students)
# Delete all elements
students.clear()
print("After clearing:", students)
print("Is empty:", len(students) == 0)

File: src/ca/qc/vc/week3/lesson8/ArrayListCRUDExample.java
package ca.qc.vc.week3.lesson8;

import java.util.ArrayList;

public class ArrayListCRUDExample {
    public static void main(String[] args) {
        ArrayList<String> students = new ArrayList<>();
        // Create
        students.add("Alice");
        students.add("Bob");
        students.add("Charlie");
        students.add(1, "David");
        System.out.println("After adding: " + students);

        // Read
        String student = students.get(2);
        System.out.println("Student at index 2: " + student);
        boolean exists = students.contains("Bob");
        System.out.println("Contains Bob: " + exists);
        int index = students.indexOf("Charlie");
        System.out.println("Index of Charlie: " + index);
        System.out.println("All students:");
        for (String name : students) {
            System.out.println(name);
        }

        // Update
        students.set(1, "Emily");
        System.out.println("After updating: " + students);

        // Delete by value
        students.remove("Bob");
        System.out.println("After removing Bob: " + students);
        // Delete by index
        students.remove(0);
        System.out.println("After removing index 0: " + students);
        // Delete all elements
        students.clear();
        System.out.println("After clearing: " + students);
        System.out.println("Is empty: " + students.isEmpty());
    }
}